kitsune/sumo/middleware.py

```diff
@@ -70,6 +73,52 @@ def process_request(self, request):
             return super(SUMORefreshIDTokenAdminMiddleware, self).process_request(request)
 
 
+class ValidateAccessTokenMiddleware(SessionRefresh):
+    """Validate the ID Token from FxA every hour."""
+
+    def __init__(self, get_response=None):
+        if not settings.OIDC_ENABLE or settings.DEV:
+            raise MiddlewareNotUsed
+        super(ValidateAccessTokenMiddleware, self).__init__(get_response=get_response)
+
+    @staticmethod
+    def get_settings(attr, *args):
+        """Override settings for Firefox Accounts."""
+        val = get_oidc_fxa_setting(attr)
+        if val is not None:
+            return val
+        return super(ValidateAccessTokenMiddleware, ValidateAccessTokenMiddleware).get_settings(
+            attr, *args
+        )
+
+    def process_request(self, request):
+        """Check the validity of the id_token every hour"""
+        if not self.is_refreshable_url(request):
+            return
+
+        # the oidc_id_token_expiration key is set by the mozilla-django-oidc library
+        expiration = request.session.get("oidc_id_token_expiration", 0)
+        now = time.time()
+        access_token = request.session.get("oidc_access_token")
+
+        if expiration < now:
+
+            response_token_info = (
+                requests.post(settings.FXA_VERIFY_URL, data={"token": access_token})
+            ).json()
+
+            # if the token is not verified, log the user out
+            if (
+                response_token_info.get("code") == 400
+                and response_token_info.get("message") == "Invalid token"
+            ):
```

Comment:
I guess here we need to also remove the `refresh_token` and double check that any tasks that depend on it first check for its existence